src/ralph/virtual/cloudsync.py

```diff
@@ -0,0 +1,75 @@
+import json
+import logging
+import threading
+
+import pkg_resources
+from django.http import (
+    HttpResponse,
+    HttpResponseBadRequest,
+    HttpResponseNotFound
+)
+from django.views.decorators.csrf import csrf_exempt
+from django.views.decorators.http import require_POST
+
+from ralph.virtual.models import CloudProvider
+
+logger = logging.getLogger(__name__)
+
+
+ENTRY_POINTS_GROUP = 'ralph.cloud_sync_processors'
+CLOUD_SYNC_PROCESSORS = None
+LOCK = threading.Lock()
+
+
+def load_processors():
+    global CLOUD_SYNC_PROCESSORS
+
+    # NOTE(romcheg): Check if processors were loaded prior to locking anything.
+    if CLOUD_SYNC_PROCESSORS is not None:
+        return
+
+    with LOCK:
+        # NOTE(romcheg): Double check in order to avoid race conditions.
+        if CLOUD_SYNC_PROCESSORS is not None:
+            return
+
+        logger.info('Loading cloud sync processors.')
+
+        CLOUD_SYNC_PROCESSORS = {}
+
+        for ep in pkg_resources.iter_entry_points(ENTRY_POINTS_GROUP):
+            try:
+                CLOUD_SYNC_PROCESSORS[ep.name] = ep.resolve()
+            except ImportError:
+                logger.error(
+                    'Could not import DC asset event processor from {}.'
+                    ''.format(ep.module_name)
+                )
+
+
+@csrf_exempt
+@require_POST
+def cloud_sync_router(request, cloud_provider_id):
+    load_processors()
```

Comment:
can't we load it in `AppConfig.ready()` or sth like that?

Reply:
Using AppConfig.ready() is bound with a few issues. One of them is that it's not even possible to import any object on a module level. In order to avoid that I have specifically re-written the patch to not use ready(). Loading the list of entry points on the first request won't hurt at all because it's won't bring any notable performance problems.